### PR TITLE
Yapf linting

### PR DESCRIPTION
**Pull request by pcmoritz**
This is linting the Python codebase with yapf and automatically checking on travis if files have linting errors. Currently excludes rllib and dataframes.

### CODE REVIEW

**Review by concretevitamin — APPROVED** [review]
LGTM feel free to merge if tests pass

File: .travis/yapf.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
```
Comment (robertnishihara) [inline .travis/yapf.sh]: Let's lint every file and not just the files modified in the commit.

**Comment by robertnishihara** [comment]
@richardliaw note that this currently does linting for Tune (but not RLlib or Pandas on Ray). Is now a reasonable time for tune?

**Comment by richardliaw** [comment]
Tune is fine right now; thanks for checking!

**Comment by Alok** [comment]
I'd tweak the style file to 

1. split before the first argument on long function calls since it makes it easy to quickly scan the argument list (`split_before_first_argument = True`)
2. not join lines to make single line if statements (breaks visual flow since you usually expect the indent) (`join_multiple_lines = False`)
3. always split if comma terminated since the clean diffs created by trailing commas only works if there's one item per line (`split_arguments_when_comma_terminated = True`)

But those are just minor nitpicks. Glad to see this.

**Pull request merged**